bluebubbles-server/src/server/databases/imessage/listeners/outgoingMessageListener.ts

```diff
@@ -160,24 +155,20 @@ export class OutgoingMessageListener extends ChangeListener {
             const cacheName = getCacheName(entry);
 
             // Skip over any that we've finished
-            if (this.cache.find(cacheName)) return;
+            if (this.cache.find(cacheName)) continue;
 
             // Add to cache
             this.cache.add(cacheName);
 
             // Reject the corresponding promise
             const idx = Server().messageManager.findIndex(entry);
             if (idx >= 0) {
-                Server().messageManager.promises[idx].reject(entry);
-
-                setTimeout(() => {
-                    super.emit("message-send-error", entry);
-                }, 1000);
+                // After the MessageMatchEmit event reaches client, then emmit message-send-error to fcm
+                await Server().messageManager.promises[idx].reject(entry);
+                super.emit("message-send-error", entry);
             } else {
+                // If it's not associated with a promise, emit it as normal
                 super.emit("message-send-error", entry);
```

Comment:
Since it now doesn't need to wait 1 second, we can move the call to emit to outside of the `idx >= 0` condition